src/components/ProductListing.js

```diff
@@ -0,0 +1,79 @@
+import React, { useReducer } from 'react'
+import { ProductsDB } from "./ProductsDB";
+import ProductCard from "./ProductCard";
+
+import "../styles.css";
+
+export default function ProductListing() {
+
+    function filterReducer(state, action) {
+        switch (action.type) {
```

Comment:
You can make a separate file for reducer to clean up the product listing page

Reply:
done